tests/pytests/unit/utils/test_files.py

```diff
@@ -0,0 +1,147 @@
+    TypeError. Python 3's open() builtin will treat the booleans as file
+    descriptor numbers and try to open stdin/stdout. We also want to test
+    fd 2 which is stderr.
+    """
+    for invalid_fn in (False, True, 0, 1, 2):
+        try:
+            with salt.utils.files.fopen(invalid_fn):
+                pass
+        except TypeError:
+            # This is expected. We aren't using an assertRaises here
+            # because we want to ensure that if we did somehow open the
+            # filehandle, that it doesn't remain open.
+            pass
+        else:
+            # We probably won't even get this far if we actually opened
+            # stdin/stdout as a file descriptor. It is likely to cause the
+            # integration suite to die since, news flash, closing
+            # stdin/stdout/stderr is usually not a wise thing to do in the
+            # middle of a program's execution.
+            pytest.fail(
+                "fopen() should have been prevented from opening a file "
+                "using {} as the filename".format(invalid_fn)
+            )
+
+
+def _create_temp_structure(temp_directory, structure):
+    for folder, files in structure.items():
+        current_directory = os.path.join(temp_directory, folder)
+        os.makedirs(current_directory)
+        for name, content in files.items():
+            path = os.path.join(temp_directory, folder, name)
+            with salt.utils.files.fopen(path, "w+") as fh:
+                fh.write(content)
+
+
+def _validate_folder_structure_and_contents(target_directory, desired_structure):
+    for folder, files in desired_structure.items():
+        for name, content in files.items():
+            path = os.path.join(target_directory, folder, name)
+            with salt.utils.files.fopen(path) as fh:
+                assert fh.read().strip() == content
+
+
+def test_recursive_copy(tmp_path):
+    src = str(tmp_path / "src")
+    dest = str(tmp_path / "dest")
+    src_structure = {
+        "foo": {"foofile.txt": "fooSTRUCTURE"},
+        "bar": {"barfile.txt": "barSTRUCTURE"},
+    }
+    dest_structure = {
+        "foo": {"foo.txt": "fooTARGET_STRUCTURE"},
+        "baz": {"baz.txt": "bazTARGET_STRUCTURE"},
+    }
+
+    # Create the file structures in both src and dest dirs
+    _create_temp_structure(src, src_structure)
+    _create_temp_structure(dest, dest_structure)
+
+    # Perform the recursive copy
+    salt.utils.files.recursive_copy(src, dest)
+
+    # Confirm results match expected results
+    desired_structure = copy.copy(dest_structure)
+    desired_structure.update(src_structure)
+    _validate_folder_structure_and_contents(dest, desired_structure)
+
+
+def test_case_sensitive_filesystem_false():
+    """
+    Test case sensitive util.
+    """
+    with patch("salt.utils.files.case_sensitive_filesystem", return_value=False):
+        result = salt.utils.files.case_sensitive_filesystem()
```

Comment:
This is not testing the new code. It patches the function to return false and just tests that the patch is working. This also goes for the test `test_case_sensitive_filesystem_true`. Can we call the function and depending on the OS we are running the test on return what we expect. So, for on windows return true and linux false?

Reply:
Can we remove these two tests now that you added the windows, linux and darwin tests below?